### PR TITLE
refactor: rpc: Pass CBlockIndex by reference instead of pointer

### PR DESCRIPTION
**Pull request by maflcko**
Follow-up to https://github.com/bitcoin/bitcoin/pull/29003#issuecomment-1841435462

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -418,7 +416,7 @@ static RPCHelpMan getdifficulty()
 {
     ChainstateManager& chainman = EnsureAnyChainman(request.context);
     LOCK(cs_main);
-    return GetDifficulty(chainman.ActiveChain().Tip());
+    return GetDifficulty(*CHECK_NONFATAL(chainman.ActiveChain().Tip()));
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/rpc/blockchain.cpp:419]: Right, you only added checks to the lines you already were touching, so I guess that makes sense.

**Review by sedited — APPROVED** [review]
ACK fa5989d514d246e56977c528b2dd2abe6dc8efcc

**Review by pablomartin4btc — APPROVED** [review]
tACK fa5989d514d246e56977c528b2dd2abe6dc8efcc

I've re-tested #29003 & `getblockchaininfo` on `mainnet`.

**Review by dergoegge — APPROVED** [review]
Code review ACK fa5989d514d246e56977c528b2dd2abe6dc8efcc

**Pull request merged**